test/atm/global-workflow/jjob_post.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -x
+bindir=$1
+srcdir=$2
+
+# prepare stuff from previous cycle
+
+# run jjob
+export HOMEgfs=$srcdir/../../ # TODO: HOMEgfs had to be hard-coded in config
+echo $HOMEgfs
+
+export EXPDIR=$bindir/test/atm/global-workflow/testrun/experiments/gdas_test
+export PDY=20210323
+export cyc=18
+export CDATE=${PDY}${cyc}
+export ROTDIR=$bindir/test/atm/global-workflow/testrun/ROTDIRS
+export CDUMP=gdas
+export DATAROOT=$bindir/test/atm/global-workflow/testrun/RUNDIR
+export COMIN_GES=${bindir}/test/atm/bkg
+export pid=${pid:-$$}
+export jobid=$pid
+export COMROOT=$DATAROOT
+
+if [ $machine != 'HERA' ]; then
```

Comment:
possibly a stupid question, but where is `$machine` defined? Isn't this at the root ctest level and would be before the configs are sourced?

Reply:
`machine` is in `config.base.emc.dyn` as 
```
echo "BEGIN: config.base"

# Machine environment
export machine="@MACHINE@"
```
The `config.base` created by `setup_expt.py` sets `machine` to the platform on which the script was run.

Reply:
yeah, but where is it sourced in this script? Isn't this submitted directly by ctest? on a fresh login, is $machine set? Maybe I'm being dense here

Reply:
You are right.   `config.base` is sourced from the g-w j-job but the j-job has not been executed at this point.   I added an echo `jjob_prep.sh` and ran it on Hera.  Here's what we get

```
1314: + echo 'machine is '
1314: machine is
1314: + '[' '!=' HERA ']'
1314: /scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/ctest_atm_3dvar/sorc/gdas.cd/test/atm/global-workflow/jjob_prep.sh: line 36: [: !=: unary op\
erator expected
1314: + sbatch -n 1 --qos=debug --time=00:10:00 --export=ALL --wait /scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/ctest_atm_3dvar/sorc/gdas.cd/.\
./..//jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP
1314: Submitted batch job 39259534
1/1 Test #1314: test_gdasapp_atm_jjob_atm_prep ...   Passed   10.95 sec
```

We are seeing the desired behavior for the wrong reason.

Reply:
g-w `build_all.sh` and `link_workfow.sh` execute `gfs_utils.fd/ush/detect_machine.sh` to set `MACHINE_ID`.   We could add similar logic to ATM jjob drivers.    An alternative is to add logic to extract `machine` from the `config.base` created by `setup_expt.py`.  Let me try the `config.base` approach.

Reply:
The following addition to `jjob_prep.sh` works on Hera.
```
# detemine machine from config.base
machine=$(echo `grep 'machine=' $EXPDIR/config.base | cut -d"=" -f2` | tr -d '"')
```
We wind up with
```
1314: + '[' HERA '!=' HERA ']'
1314: + sbatch -n 1 --qos=debug --time=00:10:00 --export=ALL --wait /scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/ctest_atm_3dvar/sorc/gdas.cd/../..//jobs/JGDAS_GLOBAL_ATMOS_ANALYSIS_PREP
1314: Submitted batch job 39260827
1/1 Test #1314: test_gdasapp_atm_jjob_atm_prep ...   Passed   42.18 sec
```

Let me check behavior on Orion.

Reply:
Extract `machine from `config.base` by executing the following line

`machine=$(echo `grep 'machine=' $EXPDIR/config.base | cut -d"=" -f2` | tr -d '"')`

in `jjob_post.sh`, `jjob_prep.sh`, and `jjob_run.sh`.   Change committed at 31306d2.